File: clients/windows/FastSort.Client.Windows/Core/Api/AuthService.cs
using FastSort.Client.Windows.Core.Api.Dto;

namespace FastSort.Client.Windows.Core.Api;

public sealed class AuthService
{
    private readonly ApiClient _apiClient;

    public AuthService(ApiClient apiClient)
    {
        _apiClient = apiClient;
    }

    public Task GenerateCaptchaAsync(string phone, string captchaType = "1", CancellationToken cancellationToken = default)
    {
        var body = new CaptchaRequest(phone, captchaType);
        return _apiClient.PostAsync("/app/user/generateCaptcha", body, cancellationToken);
    }

    public Task<LoginResponse> CaptchaLoginAsync(
        string phone,
        string captcha,
        string captchaType = "1",
        CancellationToken cancellationToken = default)
    {
        var body = new CaptchaLoginRequest(phone, captcha, captchaType);
        return _apiClient.PostAsync<LoginResponse>("/app/captchaLogin", body, cancellationToken);
    }

    public Task<LoginResponse> AccountLoginAsync(
        string username,
        string password,
        CancellationToken cancellationToken = default)
    {
        var body = new AccountLoginRequest(username, password, "1");
        return _apiClient.PostAsync<LoginResponse>("/app/accountLogin", body, cancellationToken);
    }

    public Task<ProfileResponse> GetProfileAsync(CancellationToken cancellationToken = default)
    {
        return _apiClient.PostAsync<ProfileResponse>("/app/user/getProfile", null, cancellationToken);
    }

    public Task LogoutAsync(CancellationToken cancellationToken = default)
    {
        return _apiClient.PostAsync("/app/logout", null, cancellationToken);
    }
}

File: clients/macos/FastSortClientMac/Sources/FastSortClientMac/Services/AuthService.swift
import Foundation

struct AuthService {
    let apiClient: APIClient

    func generateCaptcha(phone: String, captchaType: String = "1") async throws {
        let body = CaptchaRequest(phone: phone, captchaType: captchaType)
        try await apiClient.requestVoid("/app/user/generateCaptcha", body: body)
    }

    func captchaLogin(phone: String, captcha: String, captchaType: String = "1") async throws -> LoginResponse {
        let body = CaptchaLoginRequest(phone: phone, captcha: captcha, captchaType: captchaType)
        return try await apiClient.request("/app/captchaLogin", body: body)
    }

    func accountLogin(username: String, password: String) async throws -> LoginResponse {
        let body = AccountLoginRequest(username: username, password: password, captchaType: "1")
        return try await apiClient.request("/app/accountLogin", body: body)
    }

    func getProfile() async throws -> ProfileResponse {
        try await apiClient.request("/app/user/getProfile")
    }

    func logout() async throws {
        let body: EmptyBody? = nil
        try await apiClient.requestVoid("/app/logout", body: body)
    }
}
